listen_test.go

```diff
@@ -0,0 +1,40 @@
+package conn
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	tu "github.com/libp2p/go-testutil"
+	grc "github.com/whyrusleeping/gorocheck"
```

Comment:
We'll need to add this to the gx dependency list `gx import github.com/whyrusleeping/gorocheck` (and then answer `n` when it asks you if you'd like to rewrite the paths).

Reply:
Done. I'll admit I'm developing without `gx`, as it makes working cross-package a bit awkward, and I noticed CI runs without it anyway. Is that cool?

Reply:
(CI pointed out that `gorocheck` was already in `package.json`, so I dropped that commit.)

Reply:
Oh. Sorry, I thought it was new.